h2o-core/src/main/java/hex/grid/GridSearch.java

```diff
@@ -136,7 +138,15 @@ private GridSearch(Key<Grid> gkey, HyperSpaceWalker<MP, ?> hyperSpaceWalker) {
     // Install this as job functions
     return _job.start(new H2O.H2OCountedCompleter() {
       @Override public void compute2() {
-        gridSearch(grid);
+
+        if (_parallelism == 1) {
+          gridSearch(grid);
+        } else if (_parallelism > 1) {
+          parallelGridSearch(grid);
+        } else {
+          throw new IllegalArgumentException(String.format("Grid search parallelism level must be >= 0. Give value is '%d'.",
```

Comment:
error in message, the parallelsim here is expected to be >= 1
also I would move this check into constructor and have a simple if-else here

Reply:
This is for the case anyone will call this from tests. The check is on multiple places. I don't think it hurts to have it here.

Reply:
Oh, the message has been fixed.